server/src/main/java/com/uj/projects/booksplatform/user/service/PasswordResetServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.uj.projects.booksplatform.user.service;
+
+import com.uj.projects.booksplatform.user.service.email.EmailSender;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PasswordResetServiceImpl implements PasswordResetService {
+
+    private EmailSender emailSender;
+    private UserService userService;
+
+    @Autowired
+    public PasswordResetServiceImpl(EmailSender emailSender, UserService userService) {
+        this.emailSender = emailSender;
+        this.userService = userService;
+    }
+
+    @Override
+    public void resetPassword(String email) throws UserNotFoundException {
+        String newPassword;
+        try {
+            newPassword = userService.resetPassword(email);
+        } catch (UserNotFoundException e) {
+            System.out.print("Could not find user with email: " + email);
+            throw e;
+        }
+        emailSender.SendEmail("no-reply@booksreviewplatform.pl", email, "Password reset for Books Review Platform", "Your new pass: " + newPassword );
```

Comment:
Maybe it would be a good idea to separate these messages into some resources or configuration. Now it's hardcoded

Reply:
be more specific. Static class? What kind of configuration? What resources? How should be implemented, give examples, propose frameworks/libraries/patterns.

Reply:
Yes i meant static class. It's up to you what solution will be most appropriate. For me it is only important that this messages are not hardcoded

Reply:
done